master/internal/stream/system_intg_test.go

```diff
@@ -0,0 +1,260 @@
+//go:build integration
+// +build integration
+
+package stream
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/stream"
+	"github.com/determined-ai/determined/master/test/streamdata"
+)
+
+func simpleUpsert(i interface{}) interface{} {
+	return i
+}
+
+func recordDeletion(s1, s2 string) interface{} {
+	fmt.Println("recording deletion:", s1+" and "+s2)
+	return s2
+}
+
+type startupReadWriter struct {
+	data           []interface{}
+	startupMessage *StartupMsg
+}
+
+func (t *startupReadWriter) ReadJSON(data interface{}) error {
+	if t.startupMessage == nil {
+		return fmt.Errorf("startup message has been sent")
+	}
+	targetMsg, ok := data.(*StartupMsg)
+	if !ok {
+		return fmt.Errorf("target message type is not a pointer to StartupMsg")
+	}
+	targetMsg.Known = t.startupMessage.Known
+	targetMsg.Subscribe = t.startupMessage.Subscribe
+	t.startupMessage = nil
+	return nil
+}
+
+func (t *startupReadWriter) Write(data interface{}) error {
+	t.data = append(t.data, data)
+	return nil
+}
+
+func TestReadWriter(t *testing.T) {
+	startupMessage := StartupMsg{
+		Known: KnownKeySet{Trials: "1,2,3"},
+		Subscribe: SubscriptionSpecSet{
+			Trials: &TrialSubscriptionSpec{
+				ExperimentIds: []int{1},
+				Since:         0,
+			},
+		},
+	}
+
+	trw := startupReadWriter{
+		startupMessage: &startupMessage,
+	}
+
+	emptyMsg := StartupMsg{}
+	err := trw.ReadJSON(&emptyMsg)
+	require.NoError(t, err)
+	require.Equal(t, emptyMsg.Known, startupMessage.Known)
+	require.Equal(t, emptyMsg.Subscribe, startupMessage.Subscribe)
+	require.True(t, trw.startupMessage == nil)
+
+	err = trw.Write("test")
+	require.NoError(t, err)
+	require.Equal(t, 1, len(trw.data))
+	dataStr, ok := trw.data[0].(string)
+	require.True(t, ok)
+	require.Equal(t, "test", dataStr)
+}
+
+func TestStartup(t *testing.T) {
```

Comment:
I don't think I understand what TestStartup is testing that TestTrialStartup is not testing?

Reply:
yeah the tests are still in progress, but `TestTrialStartup` was what I used to have, and `TestStartup` tests the same stuff but is just the new version given the new refactor. `TestTrialStartup` tests just the startup function, but `TestStartup` will test more of the startup logic